povm_toolbox/library/povm_implementation.py

```diff
@@ -174,34 +175,38 @@ def _get_bitarray(self, data: DataBin) -> BitArray:
         return getattr(data, self.classical_register_name)
 
     @abstractmethod
-    def _counter(
+    def _povm_outcomes(
         self,
         bit_array: BitArray,
         povm_metadata: MetadataT,
         loc: int | tuple[int, ...] | None = None,
-    ) -> Counter:
+    ) -> List:
         """TODO."""
 
-    def get_counts_from_raw(
+    def get_povm_outcomes_from_raw(
         self,
         data: DataBin,
         povm_metadata: MetadataT,
         loc: int | tuple[int, ...] | None = None,
+        return_counts = True
```

Comment:
Would you say that this is preferable to having two separate methods?
Say e.g. `get_povm_outcomes_from_raw` and `get_povm_counts_from_raw`?

Reply:
I'd say having one method saves on some code duplication. If there is no conceivable way to obtain the counts without first computing the list of individual samples and the calling `Counter` on it, I think having one method is a bit cleaner.

Reply:
Well, the second function could almost be as simply as this:
```python
def get_povm_counts_from_raw(...):
    return Counter(self.get_povm_outcomes_from_raw)
```
So I don't think that there is really a lot of code duplication.

The reason I bring this up, is that I believe having arguments which alternate between two different return types of a function is quite bad code design. I know that this is quite "common" in Qiskit(-related) projects, but that does not mean I have to agree with it and implement it here :smile:

Reply:
Alright, I'll make the change. 👍

Reply:
I've refactored this into two functions now. There is some duplication required to handle the different cases of the input arguments, and I've kept the original design that returns either an array or a Counter object (list of tuples for samples) depending on the input.

Reply:
Okay thanks, I will take a look :+1: